server/apiserver/argoserver.go

```diff
@@ -159,6 +159,9 @@ func (as *argoServer) Run(ctx context.Context, port int) {
 	go func() { as.checkServeErr("httpServer", httpServer.Serve(httpL)) }()
 	go func() { as.checkServeErr("tcpm", tcpm.Serve()) }()
 	log.Infof("Argo Server started successfully on address %s", address)
+
+	browserOpenFunc("http://localhost" + address)
```

Comment:
not sure about pre-pending localhost here - we don't know the host?

Reply:
I do think it would be ok to just return the address or just the port here and have the func pre-pend "localhost"